Assignment 11

  **The correlation between the columns**

In [41]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [42]:
dataset1= pd.read_csv('general_data.csv')

In [43]:
dataset1.isnull()

Age  Attrition  BusinessTravel  Department  DistanceFromHome  \
0     False      False           False       False             False   
1     False      False           False       False             False   
2     False      False           False       False             False   
3     False      False           False       False             False   
4     False      False           False       False             False   
...     ...        ...             ...         ...               ...   
4405  False      False           False       False             False   
4406  False      False           False       False             False   
4407  False      False           False       False             False   
4408  False      False           False       False             False   
4409  False      False           False       False             False   

      Education  EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0         False           False          False       False   False  ...   
1         False           False          False       False   False  ...   
2         False           False          False       False   False  ...   
3         False           False          False       False   False  ...   
4         False           False          False       False   False  ...   
...         ...             ...            ...         ...     ...  ...   
4405      False           False          False       False   False  ...   
4406      False           False          False       False   False  ...   
4407      False           False          False       False   False  ...   
4408      False           False          False       False   False  ...   
4409      False           False          False       False   False  ...   

      NumCompaniesWorked  Over18  PercentSalaryHike  StandardHours  \
0                  False   False              False          False   
1                  False   False              False          False   
2                  False   False              False          False   
3                  False   False              False          False   
4                  False   False              False          False   
...                  ...     ...                ...            ...   
4405               False   False              False          False   
4406               False   False              False          False   
4407               False   False              False          False   
4408               False   False              False          False   
4409               False   False              False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
4405             False              False                  False   
4406             False              False                  False   
4407             False              False                  False   
4408             False              False                  False   
4409             False               True                  False   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0              False                    False                 False  
1              False                    False                 False  
2              False                    False                 False  
3              False                    False                 False  
4              False                    False                 False  
...              ...                      ...                   ...  
4405           False                    False                 False  
4406           False    

In [44]:
dataset1.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
4405    False
4406    False
4407    False
4408    False
4409    False
Length: 4410, dtype: bool

In [45]:
dataset1.drop_duplicates()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [46]:
dataset1.dropna()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4404   29        No      Travel_Rarely                   Sales   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4404                 4          3          Other              1        4405   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4404  Female  ...                 1.0      Y                18              8   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4404                 0               6.0                      2   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [47]:
dataset1.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [49]:
dataset1["Attrition"]=dataset1["Attrition"].map({"Yes":1, "No":0})

In [50]:
dataset1.head()

Age  Attrition     BusinessTravel              Department  \
0   51          0      Travel_Rarely                   Sales   
1   31          1  Travel_Frequently  Research & Development   
2   32          0  Travel_Frequently  Research & Development   
3   38          0         Non-Travel  Research & Development   
4   32          0      Travel_Rarely  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                 6          2  Life Sciences              1           1   
1                10          1  Life Sciences              1           2   
2                17          4          Other              1           3   
3                 2          5  Life Sciences              1           4   
4                10          1        Medical              1           5   

   Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0  Female  ...                 1.0      Y                11              8   
1  Female  ...                 0.0      Y                23              8   
2    Male  ...                 1.0      Y                15              8   
3    Male  ...                 3.0      Y                11              8   
4    Male  ...                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [52]:
dataset1 = dataset1.drop(['EmployeeCount', 'EmployeeID', 'StandardHours', 'Over18'], axis=1)  # removing unnecessary coloumns

In [53]:
dataset1.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'Gender', 'JobLevel', 'JobRole',
       'MaritalStatus', 'MonthlyIncome', 'NumCompaniesWorked',
       'PercentSalaryHike', 'StockOptionLevel', 'TotalWorkingYears',
       'TrainingTimesLastYear', 'YearsAtCompany', 'YearsSinceLastPromotion',
       'YearsWithCurrManager'],
      dtype='object')

In [51]:
from scipy.stats import pearsonr

## Correlation between Attrition and DistanceFromHome

In [55]:
stats,p =pearsonr(dataset1.Attrition, dataset1.DistanceFromHome)
print(stats,p)

-0.009730141010179692 0.5182860428050771


*Null Hypothesis: There is no Significant Correlation between Attrition and DistanceFromHome*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and DistanceFromHome*

*Correlation: -0.009730141010179692 P Value: 0.5182860428050771*

*P-Value >= 0.05 hence Null hypothesis is Accepted*
*There is no Significant Correlation between Attrition and DistanceFromHome*

## correlation between Attrition and Education

In [57]:
stats,p =pearsonr(dataset1.Attrition, dataset1.Education)
print(stats,p)

-0.015111167710968721 0.3157293177118575


*Null Hypothesis: There is no Significant Correlation between Attrition and Education*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and Education*

*Correlation: -0.015111167710968721 P Value: 0.3157293177118575*

*P-Value >= 0.05 hence Null hypothesis is Accepted*
*There is no Significant Correlation between Attrition and Education*

## correlation between Attrition and JobLevel

In [58]:
stats,p =pearsonr(dataset1.Attrition, dataset1.JobLevel)
print(stats,p)

-0.010289713287494923 0.4945171727183978


*Null Hypothesis: There is no Significant Correlation between Attrition and JobLevel*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and JobLevel*

*Correlation: -0.010289713287494923 P Value: 0.4945171727183978*

*P-Value >= 0.05 hence Null hypothesis is Accepted*
*There is no Significant Correlation between Attrition and JobLevel*

## correlation between Attrition and MonthlyIncome

In [59]:
stats,p =pearsonr(dataset1.Attrition, dataset1.MonthlyIncome)
print(stats,p)

-0.03117628169811501 0.03842748490600132


*Null Hypothesis: There is no Significant Correlation between Attrition and MonthlyIncome*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and MonthlyIncome*

*Correlation: -0.03117628169811501 P Value: 0.03842748490600132*

*P-Value < 0.05 hence Null Hypothesis is rejected, Accepting Alternate Hypothesis*
*There is Negative Correlation between Attrition and MonthlyIncome*

## correlation between Attrition and PercentSalaryHike

In [61]:
stats,p =pearsonr(dataset1.Attrition, dataset1.PercentSalaryHike)
print(stats,p)

0.0325325948910535 0.030743386433355353


*Null Hypothesis: There is no Significant Correlation between Attrition and PercentSalaryHike*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and PercentSalaryHike*

*Correlation: 0.0325325948910535 P Value: 0.030743386433355353* 

*P-Value < 0.05 hence Null Hypothesis is rejected, Accepting Alternate Hypothesis*
*There is Positive Correlation between Attrition and PercentSalaryHike*

## correlation between Attrition and StockOptionLevel

In [62]:
stats,p =pearsonr(dataset1.Attrition, dataset1.StockOptionLevel)
print(stats,p)

-0.006838852403261526 0.6498072937475723


*Null Hypothesis: There is no Significant Correlation between Attrition and StockOptionLevel*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and StockOptionLevel*

*Correlation: -0.006838852403261526 P Value: 0.6498072937475723*

*P-Value >= 0.05 hence Null hypothesis is Accepted*
*There is no Significant Correlation between Attrition and StockOptionLevel*

## correlation between Attrition and TrainingTimesLastYear

In [64]:
stats,p =pearsonr(dataset1.Attrition, dataset1.TrainingTimesLastYear)
print(stats,p)

-0.049430576244254974 0.0010247061915365072


*Null Hypothesis: There is no Significant Correlation between Attrition and TrainingTimesLastYear*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and TrainingTimesLastYear*

*Correlation: -0.049430576244254974 P Value: 0.0010247061915365072* 

*P-Value < 0.05 hence Null Hypothesis is rejected, Accepting Alternate Hypothesis*
There is Negative Correlation between Attrition and TrainingTimesLastYear*

## correlation between Attrition and YearsAtCompany

In [65]:
stats,p =pearsonr(dataset1.Attrition, dataset1.YearsAtCompany)
print(stats,p)

-0.13439221398997717 3.1638831224877484e-19


*Null Hypothesis: There is no Significant Correlation between Attrition and YearsAtCompany*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and YearsAtCompany*

*Correlation: -0.13439221398997717 P Value: 3.1638831224877484e-19* 

*P-Value < 0.05 hence Null Hypothesis is rejected, Accepting Alternate Hypothesis*
*There is Negative Correlation between Attrition and YearsAtCompany*

## correlation between Attrition and YearsSinceLastPromotion

In [66]:
stats,p =pearsonr(dataset1.Attrition, dataset1.YearsSinceLastPromotion)
print(stats,p)

-0.03301877514258437 0.028330336189396753


*Null Hypothesis: There is no Significant Correlation between Attrition and YearsSinceLastPromotion*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and YearsSinceLastPromotion*

*Correlation: -0.03301877514258437 P Value: 0.028330336189396753* 

*P-Value < 0.05 hence Null Hypothesis is rejected, Accepting Alternate Hypothesis*
*There is Negative Correlation between Attrition and YearsSinceLastPromotion*

## correlation between Attrition and YearsWithCurrManager

In [67]:
stats,p =pearsonr(dataset1.Attrition, dataset1.YearsWithCurrManager)
print(stats,p)

-0.15619931590162806 1.7339322652900218e-25


*Null Hypothesis: There is no Significant Correlation between Attrition and YearsWithCurrManager*  
*Alternate Hypothesis: There is Significant Correlation between Attrition and YearsWithCurrManager*

*Correlation: -0.15619931590162806 P Value: 1.7339322652896276e-25* 

*P-Value < 0.05 hence Null Hypothesis is rejected, Accepting Alternate Hypothesis*
*There is Negative Correlation between Attrition and YearsWithCurrManager*